Problem in installing Nuclear Boy

I have tried to install Nuclear Boy on my PC and it shows error message:


Error compiling Cython file:
                                       ------------------------------------------------------------
                       ...
                                      cdef moab.ErrorCode err
                                                                         cdef Range entities = Range()
                                  cdef moab.EntityType typ = entity_type
                                                                                    cdef vector[eh.EntityHandle] hvec
                                                 if as_list:
                                                                            err = self.inst.get_entities_by_type(<unsigned long> meshset,
                                                                                                             ^
                                  ------------------------------------------------------------
                  
                      pymoab/core.pyx:1548:48: no suitable method found
                                                                           
                                                                               Error compiling Cython file:
                               ------------------------------------------------------------
               ...
                              """
                                             cdef moab.ErrorCode err
                                                                                cdef Range ents = Range()
                                     cdef vector[eh.EntityHandle] hvec
                                                                                  if as_list:
                             err = self.inst.get_entities_by_handle(<unsigned long> meshset, hvec, recur)
                                                                               ^
                                                                                   ------------------------------------------------------------
                                                                   
                                                                       pymoab/core.pyx:1711:50: no suitable method found
                                            
                                                Error compiling Cython file:
                                                                                ------------------------------------------------------------
                                                                ...
                                                                               """
              cdef moab.ErrorCode err
                                                 cdef Range ents = Range()
                                                                                      cdef vector[eh.EntityHandle] hvec
                                                   if as_list:
                                                                              err = self.inst.get_entities_by_dimension(<unsigned long> meshset, dimension, hvec, recur)
                                                                 ^
                                                                      ------------------------------------------------------------
                                                      
                                                          pymoab/core.pyx:1751:53: no suitable method found
                               Traceback (most recent call last):
                                                                       File "<string>", line 2, in <module>
                                 File "<pip-setuptools-caller>", line 34, in <module>
           File "/home/dresden/NuclearBoy/nuclear-boy/moab-repo/build/pymoab/setup.py", line 53, in <module>
                                    setup(
                                                File "/home/dresden/NuclearBoy/nuclear-boy/lib/python3.10/site-packages/setuptools/__init__.py", line 153, in setup
           return distutils.core.setup(**attrs)
                                                     File "/usr/lib/python3.10/distutils/core.py", line 148, in setup
                                             dist.run_commands()
                                                                      File "/usr/lib/python3.10/distutils/dist.py", line 966, in run_commands
                                                                     self.run_command(cmd)
                File "/usr/lib/python3.10/distutils/dist.py", line 985, in run_command
              cmd_obj.run()
                                 File "/home/dresden/NuclearBoy/nuclear-boy/lib/python3.10/site-packages/setuptools/command/develop.py", line 34, in run
                                                                                self.install_for_development()
                                    File "/home/dresden/NuclearBoy/nuclear-boy/lib/python3.10/site-packages/setuptools/command/develop.py", line 114, in install_for_development
                        self.run_command('build_ext')
                                                           File "/usr/lib/python3.10/distutils/cmd.py", line 313, in run_command
                                                        self.distribution.run_command(command)
                    File "/usr/lib/python3.10/distutils/dist.py", line 985, in run_command
                  cmd_obj.run()
                                     File "/usr/lib/python3.10/distutils/command/build_ext.py", line 340, in run
                                        self.build_extensions()
                                                                     File "/usr/lib/python3.10/distutils/command/build_ext.py", line 449, in build_extensions
                                                                                     self._build_extensions_serial()
                                          File "/usr/lib/python3.10/distutils/command/build_ext.py", line 474, in _build_extensions_serial
                                                                  self.build_extension(ext)
                 File "/home/dresden/NuclearBoy/nuclear-boy/lib/python3.10/site-packages/Cython/Distutils/build_ext.py", line 122, in build_extension
                                                                             new_ext = cythonize(
                       File "/home/dresden/NuclearBoy/nuclear-boy/lib/python3.10/site-packages/Cython/Build/Dependencies.py", line 1134, in cythonize
                                                                             cythonize_one(*args)
                       File "/home/dresden/NuclearBoy/nuclear-boy/lib/python3.10/site-packages/Cython/Build/Dependencies.py", line 1301, in cythonize_one
                                                                                 raise CompileError(None, pyx_file)
                                       Cython.Compiler.Errors.CompileError: pymoab/core.pyx
               Compiling pymoab/core.pyx because it changed.
                                                                [1/1] Cythonizing pymoab/core.pyx
                     [end of output]

                                    note: This error originates from a subprocess, and is likely not a problem with pip.
                                        make[2]: *** [pymoab/CMakeFiles/pymoab-local-install.dir/build.make:71: pymoab/CMakeFiles/pymoab-local-install] Error 1
                                                                               make[1]: *** [CMakeFiles/Makefile2:1266: pymoab/CMakeFiles/pymoab-local-install.dir/all] Error 2
               make: *** [Makefile:146: all] Error 2
               
               
               
               
               
For additional information I am using Ubuntu 22.04 and I am not using Anaconda. Thank you sir if you can help, that would be very helpful to me. Thank you

--- a/install-nuclear-boy.sh
+++ b/install-nuclear-boy.sh
@@ -135,7 +135,7 @@ apt_package_list="software-properties-common \
 
 # list of python package
 pip_package_list="numpy \
-                  cython \
+                  cython<3 \
                   setuptools \
                   jinja2 \
                   progress \
@@ -151,10 +151,9 @@ setup_dependencies() {
     echo "Unfortunately, your operating system does not support apt-get."
     exit 1
   fi
-  sudo apt-get -y update
-  sudo apt-get -y install ${apt_package_list}
+  sudo apt-get update
+  sudo apt-get install ${apt_package_list} -y
   echo "Dependencies installed"
-  sleep 2
 }
 
 setup_python_env() {

